src/mmrelay/meshtastic/connection.py

```
@@ -1107,6 +1107,15 @@ def connect_iface(iface_param: Any) -> None:
                 facade.logger.error(f"Unknown connection type: {connection_type}")
                 return None
 
+            if client is None:
+                facade.logger.error(
+                    "Meshtastic %s connection path completed without a client.",
+                    connection_type,
+                )
+                raise ConnectionError(
+                    f"Meshtastic {connection_type} connection completed without a client."
+                )
```

Ruff (0.15.10)

[warning] Abstract `raise` to an inner function

(TRY301)